tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs

```diff
@@ -122,14 +125,33 @@ public async Task InitializeCardDataAsync()
 
     private async Task InitializeOperationDataAsync()
     {
-        RegisterForAllOperationMessages(DataExtractor.FillDotButtonOperations(ComputeSystem, _windowEx), DataExtractor.FillLaunchButtonOperations(ComputeSystem));
+        await _semaphoreSlimLock.WaitAsync();
+        try
+        {
+            RegisterForAllOperationMessages(DataExtractor.FillDotButtonOperations(ComputeSystem, _windowEx), DataExtractor.FillLaunchButtonOperations(ComputeSystem));
+
+            foreach (var data in await DataExtractor.FillDotButtonPinOperationsAsync(ComputeSystem))
+            {
+                if ((!data.WasPinnedStatusSuccessful) || (data.ViewModel == null))
+                {
+                    // TODO: pinned status for dev box for example fails often. So we'll log it and not show notifications so we don't overload the user with
+                    // failure notifications until the feature is fixed. We simply do not show the pinned icons in these cases since we don't know which ones
+                    // to show.

```

Comment:
Is this the 'it fails once after every 10 or 20 tries' thing you mentioned to me in chat?

Reply:
yes this is what I was telling you about.